packages/browser/src/browser/__tests__/inspector.integration.test.ts

```diff
@@ -55,7 +56,51 @@ describe('Inspector', () => {
 
     await deliveryPromise
 
-    expect(enrichedFn).toHaveBeenCalledTimes(1)
+    expect(enrichedFn).toHaveBeenCalledTimes(2)
     expect(deliveredFn).toHaveBeenCalledTimes(1)
   })
+
+  it('emits message failure event', async () => {
+    const [analytics] = await AnalyticsBrowser.load({
+      writeKey,
+    })
+
+    const triggeredFn = jest.fn()
+    const failedFn = jest.fn()
+    const deliveredFn = jest.fn()
+
+    analytics.on('dispatch_start', triggeredFn)
+    analytics.queue.on('delivery_failure', failedFn)
+    analytics.queue.on('message_delivered', deliveredFn)
+
+    await analytics.register({
+      name: 'Faulty Plugin',
```

Comment:
A typical "faulty" before plugin  would probably just throw an error, rather than a contextCancellation. I would probably add a test for it like so (setting retryQueue to false):
```ts
**
  it('emits message failure event', async () => {
    const [analytics] = await AnalyticsBrowser.load(
      {
        writeKey,
      },
      { retryQueue: false }
    )

    const triggeredFn = jest.fn()
    const failedFn = jest.fn()
    const deliveredFn = jest.fn()

    analytics.on('dispatch_start', triggeredFn)
    analytics.queue.on('delivery_failure', failedFn)
    analytics.queue.on('message_delivered', deliveredFn)

    await analytics.register({
      name: 'Faulty Plugin',
      version: '1.0.0',
      type: 'before',
      isLoaded: () => true,
      load: () => Promise.resolve(),
      track: () => {
        throw new Error('sup')
      },
    })

    await analytics.track('Faulty event')

    expect(triggeredFn).toHaveBeenCalledTimes(1)

    expect(triggeredFn).toHaveBeenCalledWith(
      expect.objectContaining({
        id: expect.any(String),
        event: expect.objectContaining({
          event: 'Faulty event',
          type: 'track',
        }),
      })
    )
    expect(failedFn).toHaveBeenCalledTimes(1)
    expect(deliveredFn).not.toHaveBeenCalled()
  })**
```

Reply:
The pipeline is asynchronous

Reply:
> The pipeline is asynchronous
If the ultimate goal is test are testing whether or not `failedFn` gets called and `deliveredFn` does not get called, I feel like they're the same, no (they have the same assertions)?

Reply:
It tests for exactly that - failed does get called and delivered does not

Both being triggered is a contradictory scenario

Reply:
> The pipeline is asynchronous

I guess I'm not following!

Reply:
Oh man! My bad, I forgot to disable `retryQueue` 🤦‍♂️

Thanks!